include/mbgl/style/conversion_impl.hpp

```diff
@@ -235,9 +235,7 @@ class Convertible {
         using Traits = ConversionTraits<T>;
         static VTable vtable = {
             [] (Storage&& src, Storage& dest) {
-                auto srcValue = reinterpret_cast<T&&>(src);
-                new (static_cast<void*>(&dest)) T(std::move(srcValue));
-                srcValue.~T();
+                new (static_cast<void*>(&dest)) T(reinterpret_cast<T&&>(src));
```

Comment:
I think this removes too much: Before, we created a new value `srcValue`, and moved `src` to it. Then we explicitly called the destructor, and the compiler added an implicit second destructor call, resulting in two invocations.

This code avoids `srcValue`, and directly moves the casted `src` into the destination. This means that `Storage&&` (aka `std::aligned_storage_t<...>`) gets destructed, but we don't actually call the destructor for `T`.

Reply:
Sorry, I made a reduction of this code in https://gist.github.com/kkaefer/97693761cf21d6f57348798c665ca8dd and found that this code is actually correct. I didn't consider that the vtable's `destroy` function will actually call the destructor after the move.

Reply:
nit: I'd still prefer `std::move(reinterpret_cast<T>(src))` as more evident.